openpype/hosts/maya/plugins/create/create_multishot_layout.py

```diff
@@ -0,0 +1,206 @@
+    label = "Multi-shot Layout"
+    family = "layout"
+    icon = "project-diagram"
+
+    def get_pre_create_attr_defs(self):
+        # Present artist with a list of parents of the current context
+        # to choose from. This will be used to get the shots under the
+        # selected folder to create the Camera Sequencer.
+
+        """
+        Todo: get this needs to be switched to get_folder_by_path
+              once the fork to pure AYON is done.
+
+        Warning: this will not work for projects where the asset name
+                 is not unique across the project until the switch mentioned
+                 above is done.
+        """
+
+        current_folder = get_folder_by_name(
+            project_name=get_current_project_name(),
+            folder_name=get_current_asset_name(),
+        )
+
+        current_path_parts = current_folder["path"].split("/")
+
+        items_with_label = []
+        # populate the list with parents of the current folder
+        # this will create menu items like:
+        # [
+        #   {
+        #       "value": "",
+        #       "label": "project (shots directly under the project)"
+        #   }, {
+        #       "value": "shots/shot_01", "label": "shot_01 (current)"
+        #   }, {
+        #       "value": "shots", "label": "shots"
+        #   }
+        # ]
+
+        # go through the current folder path and add each part to the list,
+        # but mark the current folder.
+        for part_idx in range(len(current_path_parts)):
+            label = current_path_parts[part_idx]
+            if current_path_parts[part_idx] == current_folder["name"]:
+                label = f"{current_path_parts[part_idx]} (current)"
+            items_with_label.append(
+                dict(label=label,
+                     value="/".join(current_path_parts[:part_idx + 1]))
+            )
+        # add the project as the first item
+        items_with_label.insert(
+            0, dict(label=f"{self.project_name} "
+                          "(shots directly under the project)", value=""))
+
+        return [
+            EnumDef("shotParent",
+                    default=current_folder["name"],
+                    label="Shot Parent Folder",
+                    items=items_with_label,
+                    ),
+            BoolDef("groupLoadedAssets",
+                    label="Group Loaded Assets",
+                    tooltip="Enable this when you want to publish group of "
+                            "loaded asset",
+                    default=False),
+            TextDef("taskName",
+                    label="Associated Task Name",
+                    tooltip=("Task name to be associated "
+                             "with the created Layout"),
+                    default="layout"),
+        ]
+
+    def create(self, subset_name, instance_data, pre_create_data):
+        shots = self.get_related_shots(
+            folder_path=pre_create_data["shotParent"]
```

Comment:
As far as I know we should always anticipate that the pre create data is not passed along - e.g. if it were called from a non-UI participant or alike. And wasn't it also the case that default values for pre create data wouldn't get passed along. I vaguely recall something like that.

@iLLiCiTiT what you think?

Reply:
isn't is so that you can pass `pre_create_data` even from *"headless"* mode? I guess the issue is that the structure of `pre_create_data` isn't clearly defined for individual creators - the required data is determined only in the code.